cli/src/commands/version.rs

```diff
@@ -22,14 +22,71 @@ use crate::ui::Ui;
 
 /// Display version information
 #[derive(clap::Args, Clone, Debug)]
-pub(crate) struct VersionArgs {}
+pub(crate) struct VersionArgs {
+    /// Display only the version number and nothing else.
+    #[arg(long)]
+    pub(crate) numeric_only: bool,
+
+    /// Display build information.
+    #[arg(long, conflicts_with = "numeric_only")]
+    pub(crate) details: bool,
+}
 
 #[instrument(skip_all)]
 pub(crate) fn cmd_version(
     ui: &mut Ui,
     command: &CommandHelper,
-    _args: &VersionArgs,
+    args: &VersionArgs,
 ) -> Result<(), CommandError> {
-    write!(ui.stdout(), "{}", command.app().render_version())?;
+    let base_version = command.app().get_version().unwrap();
+    let (version, git_commit) = if let Some(git_commit) = option_env!("JJ_GIT_COMMIT") {
```

Comment:
I'm not sure I'm reading this right, but it looks like some of the information is collected in `build.rs` and some is collected here. That seems a bit odd. Can we collect all of it in `build.rs`?